src/common/header/containers/index.tsx

```diff
@@ -1,32 +1,34 @@
 import React, { useEffect, useState } from "react";
-import { HEADER_THEME } from "../constants";
 import Header from "../components";
+import { IHeaderContainerProps } from "../interfaces";
+import { HEADER_THEME } from "../constants";
 
-const HeaderContainer = () => {
+const HeaderContainer: React.FC<IHeaderContainerProps> = ({
+  defaultTheme,
+  isScrolled,
+}) => {
   const colorSwitchPosition = 10;
   const [isMenuOpen, setIsMenuOpen] = useState(false);
-  const [headerTheme, setHeaderTheme] = useState(HEADER_THEME.transparent);
+  const [headerTheme, setHeaderTheme] = useState(defaultTheme);
 
   const menuToggleHandler = () => {
     setIsMenuOpen((isOpen) => !isOpen);
```

Comment:
Cant you just do a !isOpen instead of creating a arrow function? Can you just do a !isOpen instead of creating an arrow function? it seems to me that we don't actually need this arrow function